kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -266,6 +261,25 @@ RawBlock DBAdapter::getRawBlock(const BlockId &blockId) const {
 SetOfKeyValuePairs DBAdapter::lastReachableBlockDbUpdates(const SetOfKeyValuePairs &updates,
                                                           const OrderedKeysSet &deletes,
                                                           BlockId blockId) {
+  auto parentBlock = std::optional<RawBlock>{std::nullopt};
+  if (blockId > INITIAL_GENESIS_BLOCK_ID) {
+    parentBlock = getRawBlock(blockId - 1);
+  }
+
+  auto parentBlockDigestTask = std::make_shared<std::packaged_task<BlockDigest()>>([blockId, parentBlock] {
+    // Make sure the digest is zero-initialized by using {} initialization.
+    auto parentBlockDigest = BlockDigest{};
+    if (parentBlock) {
+      parentBlockDigest = computeBlockDigest(blockId - 1, parentBlock->data(), parentBlock->length());
+    }
+    return parentBlockDigest;
+  });
+
+  auto parentBlockDigestFuture = parentBlockDigestTask->get_future();
+
+  // Calculate the parent block digest while updating the tree.
+  asyncExecutor_.push([parentBlockDigestTask]() { (*parentBlockDigestTask)(); });
```

Comment:
Just a question, why do we capture parentBlockDigestTask?

Reply:
I've done it in such a way, because the function passed to `Handoff` is a `void ()` one and we want to call the task inside of it. Please let me know if there's a way to simplify that.